### PR TITLE
feat(python): add cwe-326 inadequate encryption strength rules

### PR DESCRIPTION
**Pull request by didroe**
## Description

Adds Python rules for CWE-326 Inadequate Encryption Strength

## Checklist
If this is your first time contributing please [sign the CLA](https://docs.bearer.com/contributing/)

- [ ] My rule has adequate metadata to explain its use.

### CODE REVIEW

**Comment by elsapet** [comment]
@didroe ahh, I can't select you to review this one since it's your PR - do you think it's good to go? My side I am happy with your changes ✅

File: rules/python/lang/weak_password_hash_argon2.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  remediation_message: |-
+    ## Description
+
+    Choosing a weak hashing algorithm for passwords compromises security. Argon2 has three variants: Argon2i, Argon2d, and Argon2id. Argon2id is the strongest and most recommended for password hashing because of its balanced resistance against both side-channel and GPU attack vectors.
+
+    ## Remediations
+
+    - **Do not** override the Argon2 type when implementing the argon2-cffi hashing library.
+      ```python
+        ph = PasswordHasher(Type.I) // unsafe
```
Comment (didroe) [inline rules/python/lang/weak_password_hash_argon2.yml:58]:
```suggestion
        ph = PasswordHasher(Type.I) # unsafe
```

File: rules/python/lang/weak_password_hash_argon2.yml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        hash = ph.hash(user.password) 
+      ```
+    - **Do** rely on the default Argon2 type (Argon2id) as it is the most secure. This ensures the highest level of security for password storage.
+      ```python
+        ph = PasswordHasher() // defaults to Argon2id
```
Comment (didroe) [inline rules/python/lang/weak_password_hash_argon2.yml:63]:
```suggestion
        ph = PasswordHasher() # defaults to Argon2id
```

**Review by didroe — COMMENTED** [review]
✅

File: rules/python/lang/weak_password_hash_argon2.yml (hunk 2 above)
Comment (elsapet) [reply rules/python/lang/weak_password_hash_argon2.yml]: Thanks for the catch(es)!

**Review by elsapet — APPROVED** [review]

**Pull request merged**